components/camel-consul/src/main/java/org/apache/camel/component/consul/endpoint/ConsulEventConsumer.java

```diff
@@ -33,32 +36,46 @@
 import org.apache.camel.component.consul.ConsulEndpoint;
 
 public final class ConsulEventConsumer extends AbstractConsulConsumer<EventClient> {
+    private ScheduledExecutorService scheduledExecutorService;
 
     public ConsulEventConsumer(ConsulEndpoint endpoint, ConsulConfiguration configuration, Processor processor) {
         super(endpoint, configuration, processor, Consul::eventClient);
+        this.scheduledExecutorService = endpoint.getCamelContext().getExecutorServiceManager().newSingleThreadScheduledExecutor(this, "ConsulEventConsumer");
     }
 
     @Override
     protected Runnable createWatcher(EventClient client) throws Exception {
         return new EventWatcher(client);
     }
 
+    @Override
+    protected void doStop() throws Exception {
+        scheduledExecutorService.shutdown();
```

Comment:
@Gr1f0n6x , please use ExecutorServiceManager.shutdownGraceful there

Reply:
@dmvolod, did it. Thank you for hint, didn't notice this method before